Add new flex features & data samples

Sergiodero opened the pull request:
This PR adds the following content to the Flexible services features page:

- Booking rules feature
- Predefined routes with deviation feature
- On-demand services with zones feature
- On-demand services with fixed stops
- Pre-requirements for all features (with special note for 3 features regarding booking_rules)
- Data samples for all features.

Could you take a look @tzujenchanmbd @isabelle-dr? both in terms of content and rendering if possible. Thanks!

Sergiodero committed:
Add new flex features & data samples

Adding:
- Booking rules feature
- Predefined routes with deviation feature
- On-demand services with zones feature
- On-demand services with fixed stops
- pre-requirements for all features
- Data samples for all features

isabelle-dr commented on docs/getting_started/features/flexible_services.md:
```diff
@@ -5,7 +5,281 @@ Flexible services, also called demand-responsibe services, are services that do
 
 Continuous Stops is used when riders can be picked up and/or dropped off between scheduled stops all along the route or on specific sections of it. This is defined in `routes.txt`, indicating that the rider can be picked up of dropped off the transit vehicle at any point along the vehicle’s travel path, on every trip of the route.
```
```suggestion
Continuous Stops is used when riders can be picked up and/or dropped off between scheduled stops. 
This can be specified either in `routes.txt`, indicating that riders can be picked up or dropped off at any point along the vehicle’s travel path for every trip of the route, or in `stop_times.txt` for a specific section of a route.

isabelle-dr replied:
Adding the stop_times.txt version in the feature description.

isabelle-dr commented on docs/getting_started/features/flexible_services.md:
```diff
@@ -5,7 +5,281 @@ Flexible services, also called demand-responsibe services, are services that do
+??? note "Sample Data"
+
+    <p style="font-size:16px">
+    The following sample shows two ways of representing continuous stop, the first one using routes.txt to specify a vehicle that can stop for pick-up and/or drop off at any point along the total length of the route. Secondly, using stop_times.txt to define a specific segment between certain stops where the rider can request to be picked-up and/or dropped off.
```
```suggestion
    The following sample shows two ways of representing Continuous Stop. 
The first sample shows that pickups and dropoffs are allowed at any point along route `RA`.
The second sample shows that pickups and dropoffs are allowed between the third and fifth stops of trip `AWE1`, accomplished by assigning `continuous_pickup` and `continuous_drop_off` values to `stop_sequence=3` and `stop_sequence=4`.

isabelle-dr replied:
An attempt to make the sample description text more specific to the sample provided.

isabelle-dr commented on docs/getting_started/features/flexible_services.md:
```diff
@@ -5,7 +5,281 @@ Flexible services, also called demand-responsibe services, are services that do
+
+##  Booking rules
+
+Booking rules can be used to enable users to reserve a trip within a demand-responsive. These rules outline the necessary prerequisites for successful bookings and provide contact information where users can make trip reservations. This should be used in conjunction with the On-demand services feature.
```
Are we missing something after "demand-responsive"?
Suggestion to replace "with the On-demand services feature" with "with one of the On-demand services features described below" since there are several.

Sergiodero replied:
I renamed the features in the latest commit following your comments and replacing On-Demand for DRT.

isabelle-dr commented on docs/getting_started/features/flexible_services.md:
```diff
@@ -5,7 +5,281 @@ Flexible services, also called demand-responsibe services, are services that do
+        | AWE1    |      6:23:00 |        6:23:00 | TAS004  |             4 |                 0 |                   0 |
+        | AWE1    |      6:25:00 |        6:25:00 | TAS005  |             5 |                   |                     |
+
+##  Booking rules
```
I'd recommend using capital letters for features (Booking Rules).

isabelle-dr commented on docs/getting_started/features/flexible_services.md:
```diff
@@ -5,7 +5,281 @@ Flexible services, also called demand-responsibe services, are services that do
+**Pre-requirement**: 
+
+- [Base features](/getting_started/features/base)
+- [Booking rules feature](/getting_started/features/flexible_services/#booking-rules) recommended if service requires booking
```
If, in the case of a service that requires booking, Booking rules _needs_ to be defined, I wouldn't use the term "recommended" here. I think we can remove it and simply have "Booking Rules feature if the service requires booking".

Sergiodero replied:
Updated!

isabelle-dr reviewed (approved):
Thank you for working on this @Sergiodero!
This is great, I just have a few minor comments & suggestions that I think apply to every page an not just this PR:

- I think we should use capital letters for features since they are going to be GTFS concepts to measure and require, and I'd make the names as short as possible (e. g. "Zone-based On-demand Services" instead of "On-demand services with zones", "Fixed Stops On-demand services" instead of "On-demand services with fixed stops")
- I'd use consistently "The following sample _shows_" instead of "The following sample defines". The sample _shows_ a feature being _defined_ via the use of certain fields and field values.

If you agree, we can address these in another PR, since it probably applies to other pages.

Sergiodero committed:
Update docs/getting_started/features/flexible_services.md

Co-authored-by: isabelle-dr <isabelle@mobilitydata.org>

Sergiodero committed:
Update docs/getting_started/features/flexible_services.md

Co-authored-by: isabelle-dr <isabelle@mobilitydata.org>

Sergiodero committed:
Address comments 2024.05.13

Changes include:

- Update samples' texts
- Rename new features with DRT language
- Change title to uppercase
- Remove "recommended" requirement

Sergiodero commented:
Thanks @isabelle-dr I added a new commit hat addresses your comments, including the samples "show" and the use of uppercases in the titles. @tzujenchanmbd do you mind having a look, specially at the samples and new features descriptions?

tzujenchanmbd commented on docs/getting_started/features/flexible_services.md:
```diff
@@ -1,11 +1,295 @@
+
+##  Booking Rules
+
+Booking rules can be used to enable users to reserve a trip on a demand-responsive service. These rules outline the necessary prerequisites for successful bookings and provide contact information where users can make trip reservations. This should be used in conjunction with the On-demand services feature.
```
....in conjunction with the **demand-responsive** services feature?

Sergiodero replied:
Good catch! I'll replace it with the text: "This feature should be used in conjunction with `Predefined Routes With Deviation`, `Zone-based Demand Responsive Services` and `Fixed-Stops Demand Responsive Services features`, if such services require booking."

tzujenchanmbd reviewed (approved):
LGTM except 1 small wording adjustment

Sergiodero commented:
Thanks @tzujenchanmbd changes have been made, you can have a look at it but for now I'll merge and if you think we should change the text we can open a new PR.

Sergiodero committed:
Address comments 2024.05.15

Corrects text in Booking Rules Description and fixes uppercasing in feature title.
